### PR TITLE
CSL 13 - PCA - Added Licence holder details and address page

### PR DESCRIPTION
**Pull request by vinodhasamiyappanHO**
## What? 
- [CSL-13](https://collaboration.homeoffice.gov.uk/jira/browse/CSL-13) - PCA - Licence holder details ('about the applicants' section)

## Why? 
- Creating new pages for precursor chemicals new applications.  Added Licence holder details and address page

## How? 
- Added necessary step configuration and field specific variables.
- Added required field validation for all pages

## Check list
- [x] I have reviewed my own pull request for linting issues (e.g. adding new lines)
- [x] I have written tests (if relevant)
- [x] I have created a JIRA number for my branch
- [x] I have created a JIRA number for my commit
- [x] I have followed the chris beams method for my commit https://cbea.ms/git-commit/
here is an [example commit](https://github.com/UKHomeOfficeForms/hof/commit/810959f391187c7c4af6db262bcd143b50093a6e)
- [x] Ensure drone builds are green especially tests
- [x] I will squash the commits before merging

### CODE REVIEW

File: apps/precursor-chemicals/sections/summary-data-sections.js
Comment (RobertMcCann) [inline apps/precursor-chemicals/sections/summary-data-sections.js:1]: I also just noticed that the addresses for this form are joined together by `,` and not \n

Comment (RobertMcCann) [reply apps/precursor-chemicals/sections/summary-data-sections.js:1]: It seems inconsistent to me actually so I'm not too sure 

Screenshot 2024-11-20 at 15 04 06

File: apps/precursor-chemicals/translations/src/en/validation.json [hunk 1]
```diff
@@ -1,31 +1,73 @@
...
+        "maxlength": "Company name must be between 2 and 200 characters",
+        "minlength": "Company name must be between 2 and 200 characters"
+    },
+    "company-number": {
+        "notUrl": "Your answer must not contain URLs or links",
+        "companyNumber": "Enter a real company number"
+    },
+    "telephone": {
+        "required": "Enter a UK telephone number",
+        "notUrl": "Your answer must not contain URLs or links",
```
Comment (dk4g) [inline apps/precursor-chemicals/translations/src/en/validation.json:14]: `notUrl` redundant, as mentioned earlier

**Review by dk4g — CHANGES_REQUESTED** [review]
there are  some indentations (4 spaces) which needs to be change to (2 spaces)

File: apps/precursor-chemicals/sections/summary-data-sections.js
Comment (dk4g) [reply apps/precursor-chemicals/sections/summary-data-sections.js:1]: let's keep it `\n` for now, I will raise this with UCD

File: apps/precursor-chemicals/fields/index.js [hunk 2]
```diff
@@ -1,4 +1,50 @@
...
+    className: ['govuk-input', 'govuk-!-width-two-thirds']
+  },
+  'company-number': {
+    mixins: ['input-text'],
+    validate: [], // additional validation covered in custom-validation.js
+    className: ['govuk-input', 'govuk-!-width-two-thirds']
+  },
+  telephone: {
+    mixins: ['input-text'],
+    validate: ['required', 'notUrl'],
```
[the remark this answers is not in the document]
Comment (dk4g) [reply apps/precursor-chemicals/fields/index.js:14]: worth adding a comment here too
`// additional validation covered in custom-validation.js`

File: apps/precursor-chemicals/translations/src/en/validation.json [hunk 3]
```diff
@@ -28,4 +69,4 @@
     "required": "Enter an email address",
     "email": "Enter a real email address"
   }
-}
+}
```
Comment (dk4g) [inline apps/precursor-chemicals/translations/src/en/validation.json:72]: No newline at end of file

**Review by dk4g — APPROVED** [review]

File: apps/precursor-chemicals/sections/summary-data-sections.js
[the remark this answers is not in the document]
Comment (vinodhasamiyappanHO) [reply apps/precursor-chemicals/sections/summary-data-sections.js:1]: Its updated

**Review by RobertMcCann — APPROVED** [review]
Looks good to me!

**Pull request merged**